src/objects/avatar/BaseAvatar.ts

```diff
@@ -17,10 +17,15 @@ import { IHitDetection } from "../../interfaces/IHitDetection";
 import { IAnimationTicker } from "../../interfaces/IAnimationTicker";
 import { Shroom } from "../Shroom";
 
+const bodyPartTypes = [ 'hd', 'bd', 'lh', 'rh' ];
+const headPartTypes = [ "hd", "fc", "ey", "hr", "hrb", "fa", "ea", "ha", "he" ];
```

Comment:
I think for both of there structures we can use a `Set` here.
Then checking if a part is in the set would be faster.

Reply:
Also, there is the `AvatarFigurePartType` enum, so instead of using `hd` for example, we could use `AvatarFigurePartType.Head`

Reply:
Thats cool, I can change these. It is quite big code source to get to grips with so I didn't actually even find that enum!